packages/clerk-js/src/core/clerk.ts

```diff
@@ -2028,8 +1993,12 @@ export class Clerk implements ClerkInterface {
       return navigateToNextStepSignUp({ missingFields: signUp.missingFields });
     }
 
-    if (this.__internal_hasAfterAuthFlows) {
-      return this.__internal_navigateToTaskIfAvailable({ redirectUrlComplete: redirectUrls.getAfterSignInUrl() });
+    if (this.session?.currentTask) {
+      await navigateIfTaskExists(this.session, {
+        baseUrl: params.signInUrl ?? displayConfig.signInUrl,
+        navigate: this.navigate,
+      });
+      return;
```

Comment:
Here - we should take into consideration `taskUrls` as well. I think it's better to just execute `setActive` in this case.